dev_tools/scripts/create_new_algorithm.sh

```diff
@@ -146,131 +154,146 @@ TEMPLATE_FILES_DESCRIPTIONS=(
+    "This is where tests with the OpenMP executor should be implemented. Usually, this means comparing against a Reference execution."
+    "This is where tests with the CUDA executor should be implemented. Usually, this means comparing against a Reference execution."
+    "This is where tests with the HIP executor should be implemented. Usually, this means comparing against a Reference execution."
 )
 
 mkdir ${TMPDIR}
 
-# create folder for temporary files
-
-# copy files needed into temporary folder
 for (( i=1; i<${#TEMPLATE_FILES[@]}+1; i++ ))
 do
     sourcename=$(echo ${TEMPLATE_FILES[$i-1]} | sed "s/${name}/${source_name}/" )
     sourcepath=${TEMPLATE_FILES_LOCATIONS[$i-1]}/${sourcename}
-    file=$(ls ${GINKGO_ROOT_DIR}/${sourcepath})
-    if [ -f "$file" ]
-    then
-        # We have evaluated the extension and found it
-        # Integrate it in the template list
-        filename=$(basename -- ${file})
-        source_path=${TEMPLATE_FILES_LOCATIONS[$i-1]}/${filename}
-        TEMPLATE_FILES[$i-1]=$(echo "${filename}" | sed "s/${source_name}/${name}/")
-    else
-        echo "Warning: Source file $sourcepath was not found."
-    fi
 
-    destpath=${TEMPLATE_FILES_LOCATIONS[$i-1]}/${TEMPLATE_FILES[$i-1]}
+    # create folder for temporary files
     mkdir -p ${TMPDIR}/${TEMPLATE_FILES_LOCATIONS[$i-1]}
-    cp ${GINKGO_ROOT_DIR}/$sourcepath ${TMPDIR}/$destpath
-done
 
-# search and replace keywords with new solver name
-echo -e "\nCreating temporary files:"
-for (( i=1; i<${#TEMPLATE_FILES[@]}+1; i++ ))
-do
-    destpath=${TEMPLATE_FILES_LOCATIONS[$i-1]}/${TEMPLATE_FILES[$i-1]}
-    perl -pi -e "s/${source_name}/$name/g" ${TMPDIR}/$destpath
-    perl -pi -e "s/${source_name^}/$Name/g" ${TMPDIR}/$destpath
-    perl -pi -e "s/${source_name^^}/$NAME/g" ${TMPDIR}/$destpath
+    # Evaluate the extension and try to find the matching files
+    for j in $(ls ${GINKGO_ROOT_DIR}/${sourcepath})
+    do
+        if [ -f "$j" ]
+        then
+            filename=$(basename -- ${j})
+            source_path=${TEMPLATE_FILES_LOCATIONS[$i-1]}/${filename}
+            destname=$(echo "${filename}" | sed "s/${source_name}/${name}/")
+            destpath=${TEMPLATE_FILES_LOCATIONS[$i-1]}/$destname
 
+            cp ${GINKGO_ROOT_DIR}/$source_path ${TMPDIR}/$destpath
 
-    # Comment all code
-    awk '/^{$/,/^}$/ { if ($0 == "{"){ print "GKO_NOT_IMPLEMENTED;"; print "//" $0; print "// TODO (script): change the code imported from '${source_type}'/'${source_name}' if needed"; next} else { print "//" $0; next }} 1' ${TMPDIR}/$destpath > tmp
-    mv tmp  ${TMPDIR}/$destpath
+            # Replace all instances of source_name by the user's requested name
+            perl -n -i -e "print unless m/.*common.*${source_name}_kernels.hpp.inc.*/" ${TMPDIR}/$destpath
+            perl -pi -e "s/${source_name}/$name/g" ${TMPDIR}/$destpath
+            perl -pi -e "s/${source_name^}/$Name/g" ${TMPDIR}/$destpath
```

Comment:
Do we not need `$NAME` any more?

Reply:
good point, I don't know what happened to that one, I will put it back